services/spar/test/Test/Spar/Scim/UserSpec.hs

```diff
@@ -99,77 +97,45 @@ type EffsWithoutBrigAccess =
      Final IO
    ]
 
-type Effs = BrigAccess ': EffsWithoutBrigAccess
-
-type InterpreterState =
-  ( Map (Data.Id.TeamId, Wire.API.User.Identity.Email) Data.Id.UserId,
-    ( Map Data.Id.UserId (Data.Json.Util.UTCTimeMillis, Data.Json.Util.UTCTimeMillis),
-      ( Map UserRefOrd UserId,
-        (Spar.Sem.IdPConfigStore.Mem.TypedState, Either ScimError ())
-      )
-    )
-  )
-
-handlerResult :: InterpreterState -> Either ScimError ()
-handlerResult = snd . snd . snd . snd
-
 interpretWithBrigAccessMock ::
-  ( Sem Effs (Either ScimError ()) ->
-    Sem EffsWithoutBrigAccess (Either ScimError ())
+  ( Sem (BrigAccess ': EffsWithoutBrigAccess) a ->
+    Sem EffsWithoutBrigAccess a
   ) ->
-  Sem Effs (Either ScimError ()) ->
-  IO InterpreterState
+  Sem (BrigAccess ': EffsWithoutBrigAccess) a ->
+  IO a
 interpretWithBrigAccessMock mock =
   runFinal
     . embedToFinal @IO
     . discardTinyLogs
-    . scimExternalIdStoreToMem
-    . scimUserTimesStoreToMem
-    . samlUserStoreToMem
-    . idPToMem
+    . ignoringState scimExternalIdStoreToMem
+    . ignoringState scimUserTimesStoreToMem
+    . ignoringState samlUserStoreToMem
+    . ignoringState idPToMem
```

Comment:
https://github.com/isovector/wire-server/pull/1

Reply:
This is a plausible approach, but it's less composable, because you need to write a custom `ignoringState` for every shape of resultant composed interpreters.